GHA Windows testing is broken

Unfortunately #158 added conditional initialization on GHA broke Windows testing:

```
Run if [ `tox list --no-desc -f init|wc -l` = 1 ]; then tox -e init;else true; fi
ParserError: D:\a\_temp\e9225f41-01a8-4d2c-bdae-00a5b23c7645.ps1:2
Line |
   2 |  if [ `tox list --no-desc -f init|wc -l` = 1 ]; then tox -e init;else  …
     |    ~
     | Missing '(' after 'if' in if statement.

Error: Process completed with exit code 1.
```

See it on https://github.com/plone/Products.MimetypesRegistry/actions/runs/6517361159/job/17701765170

I have no Windows machine around to try a similar conditional for it, meanwhile we could specify that the OS needs to be Linux to run this?

As far as I'm aware, we only do Windows tests on Products.MimeTypesRegistry only? 🤔 and on that one, so far, we are not using `mxdev` and for that there is no `tox init` to run 🍀

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -38,6 +38,8 @@ jobs:
       - name: Install dependencies
         run: python -m pip install tox
       - name: Initialize tox
+        # the bash one-liner below does not work on Windows
+        if: contains(matrix.os, 'ubuntu')
         run: |
           if [ `tox list --no-desc -f init|wc -l` = 1 ]; then tox -e init;else true; fi
       - name: Run tests

--- a/news/180.bugfix
+++ b/news/180.bugfix
@@ -1,0 +1,2 @@
+Only run a GHA on Linux, while we wait for a counterpart for Windows.
+[gforcada]

